src/server.rs

```diff
@@ -47,13 +49,41 @@ impl Server for RocketServer {
     }
 }
 
-/// Searches a single item
-#[get("/search?<query>")]
+/// The selected search scope
+#[derive(Debug)]
+pub enum Scope {
+    /// Search by name and alternative names
+    Names,
+    /// Search only e numbers
+    ENumber,
+}
+
+impl<'v> FromFormValue<'v> for Scope {
+    type Error = &'v RawStr;
+
+    fn from_form_value(form_value: &'v RawStr) -> Result<Self, &'v RawStr> {
+        let decoded_value = form_value.percent_decode().map_err(|_| form_value)?;
+        match decoded_value.as_ref() {
+            "names" => Ok(Scope::Names),
+            "eNumber" => Ok(Scope::ENumber),
+            _ => Err(form_value),
+        }
+    }
+}
+
+/// Searches a single item with a scope
+#[get("/search?<query>&<scope>")]
 fn search(
     query: String,
+    scope: Option<Scope>,
```

Comment:
Are you sure we should allow omitting the scope?

Reply:
67b5841